scripts/bump-edge.ts

```diff
@@ -3,18 +3,23 @@ import { $fetch } from 'ofetch'
 import { inc } from 'semver'
 import { determineBumpType, loadWorkspace } from './_utils'
 
+const nightlyPackages = {
+  nitropack: 'nitropack-edge',
+  h3: 'h3-nightly',
+  nuxi: 'nuxi-ng'
+}
+
 async function main () {
   const workspace = await loadWorkspace(process.cwd())
 
   const commit = execSync('git rev-parse --short HEAD').toString('utf-8').trim().slice(0, 8)
   const date = Math.round(Date.now() / (1000 * 60))
 
-  const nuxtPkg = workspace.find('nuxt')
-  const { version: latestNitro } = await $fetch<{ version: string }>('https://registry.npmjs.org/nitropack-edge/latest')
-  nuxtPkg.data.dependencies.nitropack = `npm:nitropack-edge@^${latestNitro}`
-
-  const { version: latestNuxi } = await $fetch<{ version: string }>('https://registry.npmjs.org/nuxi-ng/latest')
-  nuxtPkg.data.dependencies.nuxi = `npm:nuxi-ng@^${latestNuxi}`
+  const resolutions: Record<string, string> = {}
+  for (const [pkg, name] of Object.entries(nightlyPackages)) {
+    const { version: latest } = await $fetch<{ version: string }>(`https://registry.npmjs.org/${name}/latest`)
+    resolutions[pkg] = `npm:${name}@^${latest}`
```

Comment:
I am thinking to use `h3-nightly@latest` (same for nitropack) to remove requirement of two time releases nitro>nuxt (and if both are not released we might have conflicting h3 nightly versions by each)

Reply:
A couple of issues:

* see [pnpm resolution mode](https://pnpm.io/npmrc#resolution-mode) - default behaviour best case is that `latest` will be resolved to be latest version when package was published (same as hash here)
* in some cases it can be the latest version _already known about_ on user computer, ie. not actually latest. by using caret + hash we _force_ update to at least that version, but allow it to be newer

Reply:
I am not sure if Caret is respected for edge versions with `-` suffix. Also main issue is that nitro and nuxt requesting different nightly version of h3 (this is issue as of today since nitro requires h3 with `^1.8.0-rc.3` but that is also not respected. Also notice that we should use `"h3": "npm:..."` directly in dependencies.

Reply:
> Also notice that we should use "h3": "npm:..." directly in dependencies.

What do you mean?

Reply:
On other solution would be that nuxt pulls out the `h3` version from latest `nitropack-edge` version by the time of publishing nuxt-edge but it is really not good honestly because everytime i want to rest nitropack, i have to manually add nitropack resolution too.

(avoiding `latest` was my initial intention that wrote this script with fetch.. but it proven to be less effective)

Reply:
The caret is respected for edge versions: see https://stackblitz.com/edit/github-fwjf8z?file=package.json,pnpm-lock.yaml

Reply:
I still think latest is the right choice because technically versioning (endge edge versioning) of nuxt, nitropack and h3 is decoupled. I hope you respect this decision from the parson that is mainly maintaining both of them. This helps me much better in triage and when asking people to use nuxt-edge instead of additional steps for h3, nitropack additional resolution.

Reply:
have you tested using `latest` in nuxt package? I don't think it will solve this issue.

Reply:
I (re)tested with an older caret h3 in nitro. pnpm resolves to exactly the same nightly version (it is a valid behavior from pnpm. versions with `-` suffix are exact match) not latest (by the time f3d0bc9)

<img width="499" alt="image" src="https://github.com/nuxt/nuxt/assets/5158436/9bcfd534-6b9b-4d39-a783-f5608cfef656">

in order to avoid hosting, both nitro-edge and nuxt3 need same version range request. And (sadly) `^1` is not also possible with edge:

<img width="710" alt="image" src="https://github.com/nuxt/nuxt/assets/5158436/ded32983-8b2d-4760-860f-c616b27441c2">

Reply:
Okay - let's give it a try. I think pnpm will resolve to latest version as at time of when the nightly version was published but hopefully if the constraints are the same that will avoid different edge versions 🤞

Reply:
Sure! we might give also dashless naming (h3-nightly@1.8.1001) a try later on to keep on semver-major at least but appreciated moving this forward 🙏🏼